lib/ansible/modules/cloud/vmware/vmware_vsan_health_facts.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/python
+
+# (c) 2019, OVH SAS
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: vmware_vsan_health_facts
+
+short_description: Gather facts about a VMware vSAN cluster's health.
+
+version_added: "2.9"
+
+description:
+    - "Gather facts about a VMware vSAN cluster's health"
+
+options:
+    cluster_name:
+        description:
+            - Name of the vSAN cluster
+        required: true
+        type: string
+    fetch_from_cache:
+        description:
+            - True to return the result from cache directly instead of running the full health check.
+        required: false
+        default: false
+        type: bool
+
+extends_documentation_fragment:
+    - vmware.documentation
+
+author:
+    - Erwan Quelin (@equelin)
+'''
+
+EXAMPLES = '''
+- name: Gather health facts from a vSAN's cluster
+  hosts: localhost
+  gather_facts: 'no'
+  vmware_vsan_health_facts:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    cluster_name: 'vSAN01'
+    fetch_from_cache: 'False'
+  register: facts
+'''
+
+RETURN = '''
+health:
+    description: vSAN cluster health facts
+    type: dict
+    sample: {
+        "_vimtype": "vim.cluster.VsanClusterHealthSummary",
+        "burnInTest": null,
+        "clusterStatus": {
+            "_vimtype": "vim.cluster.VsanClusterHealthSystemStatusResult",
+            "goalState": "installed",
+            "status": "green",
+            "trackedHostsStatus": [
+                {
+                    "_vimtype": "vim.host.VsanHostHealthSystemStatusResult",
+                    "hostname": "esxi01.example.com",
+                    "issues": [],
+                    "status": "green"
+                },
+                {
+                    "_vimtype": "vim.host.VsanHostHealthSystemStatusResult",
+                    "hostname": "esxi04.example.com",
+                    "issues": [],
+                    "status": "green"
+                },
+                {
+                    "_vimtype": "vim.host.VsanHostHealthSystemStatusResult",
+                    "hostname": "esxi02.example.com",
+                    "issues": [],
+                    "status": "green"
+                },
+                {
+                    "_vimtype": "vim.host.VsanHostHealthSystemStatusResult",
+                    "hostname": "esxi03.example.com",
+                    "issues": [],
+                    "status": "green"
+                }
+            ],
+            "untrackedHosts": []
+        }
+    }
+'''
+
+try:
+    from pyVmomi import vim, vmodl, VmomiSupport
+    HAS_PYVMOMI = True
+    HAS_PYVMOMIJSON = hasattr(VmomiSupport, 'VmomiJSONEncoder')
+except ImportError:
+    PYVMOMI_IMP_ERR = traceback.format_exc()
```

Comment:
I think you missed `import traceback` 
ERROR: lib/ansible/modules/cloud/vmware/vmware_vsan_health_facts.py:103:22: undefined-variable Undefined variable 'traceback'

Reply:
Done